tinker_cookbook/renderers.py

```diff
@@ -324,11 +540,17 @@ def __init__(self, tokenizer: Tokenizer):
         self.tokenizer = tokenizer
 
     def _preprocess_message_parts(self, message: Message) -> list[ImagePart | TextPart]:
-        return (
-            message["content"]
-            if isinstance(message["content"], list)
-            else [TextPart(type="text", text=message["content"])]
-        )
+        """Convert message content to list form, filtering out ThinkingPart and ToolCallPart.
+
+        ThinkingPart and ToolCallPart are filtered because most renderers
+        (especially VL renderers) don't support them in the message parts.
+        """
```

Comment:
This is a strange method to have on the base class. why is it here? Is it just for the VL renderer? If so, why not move it there?